src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -654,5 +656,37 @@ public static BodyPublisher ofByteArrays(Iterable<byte[]> iter) {
         public static BodyPublisher noBody() {
             return new RequestPublishers.EmptyPublisher();
         }
+
+        /**
+         * Creates a {@code BodyPublisher} that publishes a request
+         * body consisting of the aggregation of the bytes published
+         * by a sequence of publishers.
+         *
+         * <p> If the sequence is empty an {@linkplain #noBody() empty} publisher
+         * is returned. Otherwise each publisher is lazily subscribed to in turn,
+         * until all the body bytes are published, an error occurs, or the
+         * returned publisher's subscription is cancelled.
+         *
+         * <p> The request body published by the aggregate body publisher
+         * is logically equivalent to the request body that would have
+         * been published by aggregating all the bytes of each publisher
+         * in sequence. However, an implementation is free to split
+         * the sequence of bytes in any way it chooses.
+         *
+         * @implNote If the aggregate
+         * publisher's subscription is {@linkplain Flow.Subscription#cancel()
+         * cancelled}, or an error occurs while publishing the bytes, not all
+         * publishers in the sequence may be subscribed to.
+         * A publisher may be subscribed several times in sequence if the
+         * request needs to be resent.
+         *
+         * @param publishers a sequence of publishers.
+         * @return An aggregate publisher that publishes a request body
+         * logically equivalent to the concatenation of all bytes published
+         * by each publisher in the sequence.
+         */
+        public static BodyPublisher concat(BodyPublisher... publishers) {
```

Comment:
It's a choice whether to allow an empty publishers array, or an array of just 1, or to disallow (throw IllegalArgumentException), but that will depend on expected usage.

Reply:
Yes - returning `noBody()` seems more natural - since varargs allows you to call `BodyPublishers.concat()` with no parameters. I toyed with the idea of defining the method as `BodyPublisher concat(BodyPublisher first, BodyPublisher... rest)` but that's very cumbersome for the case where you already have a `List<BodyPublishers>` or a `BodyPublisher[]` array.